File: apps/qizhi/mobile/MentorAI-iOS/MentorAI/Authentication/AuthConfig.swift
import Foundation

/// Which backend environment this build targets. The test (internal/QA) environment defaults
/// to the test server and shows the "暂不登录" shortcut; production uses the ZJU server and
/// hides it. Driven by build configuration — flip the values below to force an environment.
enum AppEnvironment {
    case test
    case production

    #if DEBUG
    static let current: AppEnvironment = .test
    #else
    static let current: AppEnvironment = .production
    #endif

    var config: AuthConfig {
        switch self {
        case .test:       return .test
        case .production: return .production
        }
    }

    /// Whether to show the "暂不登录" (skip-login / 模拟登录) button on the login screen.
    var showsSkipLogin: Bool { self == .test }
}

struct AuthConfig {
    var apiBaseURL: URL

    static let production = AuthConfig(
        apiBaseURL: URL(string: "http://jsfzai.zju.edu.cn/api")!
    )

    static let test = AuthConfig(
        apiBaseURL: URL(string: "http://127.0.0.1:8000")!
    )

    /// Starts on the build's target environment; `switchToTest`/`switchToProduction` change it
    /// at runtime as the user picks a login method (see `AppState.skipLogin` / `signIn`).
    private static var _active: AuthConfig = AppEnvironment.current.config

    static var `default`: AuthConfig {
        get { _active }
        set { _active = newValue }
    }

    static func switchToTest() {
        _active = .test
    }

    static func switchToProduction() {
        _active = .production
    }
}

File: apps/qizhi/mobile/MentorAI-Android/app/src/main/kotlin/com/mentorai/app/authentication/AuthConfig.kt
package com.mentorai.app.authentication

/**
 * Which backend environment this build targets. The test (internal/QA) environment uses the
 * test server and shows the "暂不登录" shortcut; production uses the ZJU server and hides it.
 * Production-only: every build (debug + release) targets Production and hides 暂不登录 (see
 * `current` below). Mirrors the iOS `AppEnvironment`; Android wires one base URL at app
 * construction rather than switching per login action.
 */
enum class AppEnvironment {
    Test,
    Production;

    val config: AuthConfig
        get() = when (this) {
            Test -> AuthConfig.Test
            Production -> AuthConfig.Production
        }

    /** Whether to show the "暂不登录" (skip-login / 模拟登录) button on the login screen. */
    val showsSkipLogin: Boolean get() = this == Test

    companion object {
        // Production-only: every build (debug + release) targets the ZJU production backend and
        // hides 暂不登录, so real users can't accidentally land in the test database.
        // (For local testing against 127.0.0.1, temporarily change this to Test.)
        val current: AppEnvironment = Production
    }
}

/**
 * Backend base URL. The MentorAI/启智 backend runs over http; the network security config
 * whitelists *.zju.edu.cn and the test server IP for cleartext traffic.
 */
data class AuthConfig(
    val apiBaseUrl: String,
) {
    companion object {
        val Production = AuthConfig(apiBaseUrl = "http://jsfzai.zju.edu.cn/api")
        val Test = AuthConfig(apiBaseUrl = "http://127.0.0.1:8000")

        /** The active config for this build's environment. */
        val Default: AuthConfig get() = AppEnvironment.current.config
    }
}
